src/plugins/info.ts

```diff
@@ -0,0 +1,60 @@
+import type { Plugin } from 'vite'
+import type { PWAPluginContext } from '../context'
+import {
+  PWA_INFO_VIRTUAL,
+  RESOLVED_PWA_INFO_VIRTUAL,
+} from '../constants'
+import type { VitePluginPWAAPI } from '../types'
+
+export function InfoPlugin(ctx: PWAPluginContext, api: VitePluginPWAAPI) {
+  return <Plugin>{
+    name: 'vite-plugin-pwa:info',
+    enforce: 'post',
+    resolveId(id) {
+      if (id === PWA_INFO_VIRTUAL)
+        return RESOLVED_PWA_INFO_VIRTUAL
+
+      return undefined
+    },
+    load(id) {
+      if (id === RESOLVED_PWA_INFO_VIRTUAL)
+        return generatePwaInfo(ctx, api)
+    },
+  }
+}
+
+function generatePwaInfo(ctx: PWAPluginContext, api: VitePluginPWAAPI) {
+  const webManifestData = api.webManifestData()
+  if (!webManifestData)
+    return 'export const pwaInfo = undefined;'
+
+  const { href, useCredentials, toLinkTag } = webManifestData
+  const registerSWData = api.registerSWData()
+
+  const entry = {
```

Comment:
Can we expose a type for it?

Reply:
I'll add the type in main branch, I have errors when filling it, the type will be internal, just a copy paste from the virtual